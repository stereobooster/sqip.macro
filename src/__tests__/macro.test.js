const path = require("path");
const pluginTester = require("babel-plugin-tester");
const plugin = require("babel-plugin-macros");
const prettier = require("prettier");

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: {
    filename: __filename,
  },
  formatResult(result) {
    return prettier.format(result, { trailingComma: "es5" });
  },
  tests: {
    "no usage": `import sqip from '../macro'`,
    "correct usage": `
      import sqip from '../macro';

      const preview = sqip('./fixtures/jonathan-daniels-416786-unsplash.jpg');
    `,
    // "correct usage, wrong file": `
    //   import sqip from '../macro';

    //   const preview = sqip('./fixtures/readme.md');
    // `,
  },
});
