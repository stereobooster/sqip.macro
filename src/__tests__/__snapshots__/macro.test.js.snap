// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros correct usage: correct usage 1`] = `
"
import sqip from '../macro';

const preview = sqip('./fixtures/jonathan-daniels-416786-unsplash.jpg');

      ↓ ↓ ↓ ↓ ↓ ↓

const preview =
  \\"data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 4774 3024'%3e%3cfilter id='b'%3e%3cfeGaussianBlur stdDeviation='12' /%3e%3c/filter%3e%3cpath fill='%23515a57' d='M0 0h4774v3021H0z'/%3e%3cg filter='url(%23b)' transform='translate(9.3 9.3) scale(18.64844)' fill-opacity='.5'%3e%3cellipse fill='whitefef' rx='1' ry='1' transform='matrix(22.2926 -26.19371 77.82862 66.23735 156.3 108.8)'/%3e%3cellipse fill='black70d' cx='22' cy='46' rx='47' ry='102'/%3e%3cellipse fill='black80c' cx='229' cy='43' rx='75' ry='68'/%3e%3cellipse fill='whitece6' rx='1' ry='1' transform='matrix(-12.63535 -11.48194 24.2777 -26.71648 89.8 65.8)'/%3e%3cellipse fill='white' rx='1' ry='1' transform='matrix(27.96694 18.2807 -10.74726 16.44182 197.4 128.3)'/%3e%3cellipse fill='white' rx='1' ry='1' transform='rotate(13.3 -174.1 546.6) scale(28.16331 15.08078)'/%3e%3cellipse fill='%2304201e' rx='1' ry='1' transform='matrix(7.92207 18.1329 -160.92832 70.30786 129.2 6)'/%3e%3cellipse fill='%231b2129' rx='1' ry='1' transform='rotate(122.7 10 84.6) scale(34.63705 18.20046)'/%3e%3cellipse fill='%239a7453' rx='1' ry='1' transform='matrix(-27.68128 18.11412 -11.37366 -17.38077 143 124)'/%3e%3cellipse fill='%23b4b2bf' cx='23' cy='161' rx='62' ry='10'/%3e%3c/g%3e%3c/svg%3e\\";

"
`;

exports[`macros no usage: no usage 1`] = `
"
import sqip from '../macro'

      ↓ ↓ ↓ ↓ ↓ ↓


"
`;
